src/imageResource.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+require('openseadragon');
+
+var ImageResource = function(config, dispatcher) {
+  this.needed = config.needed || false;
+  this.visible = config.visible || false;
+  this.clipRegion = config.clipRegion;
+  this.opacity = config.opacity || 1;
+  this.x = config.x || 0;
+  this.y = config.y || 0;
+  this.height = config.height || 1;
+  this.width = config.width || 1;
+  this.zIndex = config.zIndex || 0;
+  this.tileSource = config.tileSource;
+  this.dynamic = config.dynamic || false;
+  this.imageType = config.imageType || "main"; // can be 'main', 'alternate', or 'detail'
+
+  this.status = 'initialized'; // can be 'requested', 'received', 'pending','shown', or 'failed'
+  this.dispatcher = dispatcher;
+};
+
+ImageResource.prototype = {
+  hide: function() {
+
+  },
+
+  setOpacity: function(opacity) {
+    this.opacity = opacity;
+    this.tiledImage.setOpacity(opacity);
+  },
+
+  getOpacity: function() {
+    return this.opacity;
+  },
+
+  openTileSource: function(viewer, parentBounds, parentHandler) {
+    var self = this;
+
+    // We've already loaded this tilesource
+    if(this.status === 'shown') {
+      return;
+    }
+
+    // otherwise, continue loading the tileSource.
+    this.dispatcher.emit('image-resource-tile-source-requested', { 'detail': this.tileSource });
+    self.status = 'requested';
+    var position = this._getPositionInViewer(parentBounds);
+    viewer.addTiledImage({
+      x: position.x,
+      y: position.y,
+      width: parentBounds.width * this.width,
```

Comment:
Maybe we need _getBoundsInViewer so we can have the width as well without having to calculate it here.